src/main/java/sg/ncl/MainController.java

```diff
@@ -334,6 +336,62 @@ public String maintainance() {
         return "maintainance";
     }
 
+    @RequestMapping(value = "/networkTool", method = RequestMethod.GET)
+    public String networkTopologyTool() {
+        return "network_diagram";
+    }
+
+    @RequestMapping(value = "/networkTool", method = RequestMethod.POST)
+    public @ResponseBody String networkTopologyAnalysis(@RequestParam("jsonText") String jsonText) {
+        String nsfilename = null;
+        JSONObject jsonObject = new JSONObject();
+        StringBuilder logBuilder = new StringBuilder();
+        String filename = networkToolProperties.getTemp() + System.currentTimeMillis() + ".json";
+        try (FileWriter fw = new FileWriter(filename)) {
+            fw.write(jsonText);
+            fw.close();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this "close" call; closing the resource is handled automatically by the try-with-resources. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4087)